beacon-chain/core/peerdas/reconstruction.go

```diff
@@ -28,6 +28,56 @@ func MinimumColumnCountToReconstruct() uint64 {
 	return (params.BeaconConfig().NumberOfColumns + 1) / 2
 }
 
+// recoverCellsForBlobs reconstructs cells for all blobs from the given data column sidecars.
+// When withProofs is true, it returns both cells and proofs. When false, it only returns cells (optimized).
+func recoverCellsForBlobs(verifiedRoSidecars []blocks.VerifiedRODataColumn, blobCount int, withProofs bool) ([][]kzg.Cell, [][]kzg.Proof, error) {
+	sidecarCount := len(verifiedRoSidecars)
+	var wg errgroup.Group
+
+	cellsPerBlob := make([][]kzg.Cell, blobCount)
+	var proofsPerBlob [][]kzg.Proof
+	if withProofs {
+		proofsPerBlob = make([][]kzg.Proof, blobCount)
+	}
+
+	for blobIndex := range uint64(blobCount) {
+		wg.Go(func() error {
+			cellsIndices := make([]uint64, 0, sidecarCount)
+			cells := make([]kzg.Cell, 0, sidecarCount)
+
+			for _, sidecar := range verifiedRoSidecars {
+				cell := sidecar.Column[blobIndex]
+				cells = append(cells, kzg.Cell(cell))
+				cellsIndices = append(cellsIndices, sidecar.Index)
+			}
+
+			if withProofs {
+				recoveredCells, recoveredProofs, err := kzg.RecoverCellsAndKZGProofs(cellsIndices, cells)
+				if err != nil {
+					return errors.Wrapf(err, "recover cells and KZG proofs for blob %d", blobIndex)
+				}
+				cellsPerBlob[blobIndex] = recoveredCells
+				proofsPerBlob[blobIndex] = recoveredProofs
+			} else {
+				recoveredCells, err := kzg.RecoverCells(cellsIndices, cells)
+				if err != nil {
+					return errors.Wrapf(err, "recover cells for blob %d", blobIndex)
+				}
+				cellsPerBlob[blobIndex] = recoveredCells
+			}
+			return nil
```

Comment:
In case my comment about 2 functions is not applied:


```suggestion
			if withProofs {
				recoveredCells, recoveredProofs, err := kzg.RecoverCellsAndKZGProofs(cellsIndices, cells)
				if err != nil {
					return errors.Wrapf(err, "recover cells and KZG proofs for blob %d", blobIndex)
				}

				cellsPerBlob[blobIndex] = recoveredCells
				proofsPerBlob[blobIndex] = recoveredProofs

                return nil
			} 

			recoveredCells, err := kzg.RecoverCells(cellsIndices, cells)
			if err != nil {
				return errors.Wrapf(err, "recover cells for blob %d", blobIndex)
			}

			cellsPerBlob[blobIndex] = recoveredCells
```

Reply:
will go with the 2 functions